core/dbt/adapters/relation_configs/config_change.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Hashable
+
+from dbt.adapters.relation_configs.config_base import RelationConfigBase
+from dbt.dataclass_schema import StrEnum
+
+
+class RelationConfigChangeAction(StrEnum):
+    alter = "alter"
+    create = "create"
+    drop = "drop"
+
+
+@dataclass(frozen=True, eq=True, unsafe_hash=True)
+class RelationConfigChange(RelationConfigBase, ABC):
+    action: RelationConfigChangeAction
+    context: Hashable  # this is usually a RelationConfig, e.g. IndexConfig, but shouldn't be limited
```

Comment:
should those both be marked as optional? `is_change` checks for `None` values for both

Reply:
Neither is optional actually. A change needs both. I'll look at what I did with `is_change` and update that then.

Reply:
I changed it to `and` instead of `or`. Ideally every instance of `RelationConfigChange` is a change, but I'm allowing for a small degree of sloppiness I guess. In other words, I can pass in a `context` of `None` and let the object determine that's not a change, instead of checking for `None` when creating the instance.